libbeat/template/template.go

```diff
@@ -210,20 +210,21 @@ func (t *Template) Generate(properties common.MapStr, dynamicTemplates []common.
 		indexSettings.Put("number_of_routing_shards", defaultNumberOfRoutingShards)
 	}
 
-	if t.esVersion.IsMajor(7) {
-		defaultFields = append(defaultFields, "fields.*")
-		indexSettings.Put("query.default_field", defaultFields)
-	}
-
-	indexSettings.DeepUpdate(t.config.Settings.Index)
-
 	var mappingName string
 	if t.esVersion.Major >= 6 {
 		mappingName = "doc"
 	} else {
 		mappingName = "_default_"
 	}
 
+	if t.esVersion.IsMajor(7) {
```

Comment:
We should convert this to >= 7 like we have on line 221. Also we should probably do the version check only once.

Current code LGTM but I think we should clean it up in a follow up or here to make sure we don't fall into the version 8 trap.

Reply:
changed logic to use `>=`